assets/js/googlesitekit/data/create-settings-store.test.js

```diff
@@ -0,0 +1,453 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+import { createRegistry } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import {
+	muteConsole,
+	subscribeUntil,
+	unsubscribeFromAll,
+} from 'tests/js/utils';
+import { createSettingsStore } from './create-settings-store';
+
+const SETTING = {
+	slug: 'testSetting',
+	action: 'setTestSetting',
+	selector: 'getTestSetting',
+	actionType: 'SET_TEST_SETTING',
+};
+const STORE_ARGS = [ 'core', 'site', 'settings' ];
+
+describe( 'createSettingsStore store', () => {
+	let apiFetchSpy;
+	let dispatch;
+	let registry;
+	let select;
+	let storeDefinition;
+	let store;
+
+	beforeAll( () => {
+		API.setUsingCache( false );
+	} );
+
+	beforeEach( () => {
+		registry = createRegistry();
+
+		storeDefinition = createSettingsStore( ...STORE_ARGS, {
+			settingSlugs: [ SETTING.slug ],
+			registry,
+		} );
+		registry.registerStore( storeDefinition.STORE_NAME, storeDefinition );
+		dispatch = registry.dispatch( storeDefinition.STORE_NAME );
+		store = registry.stores[ storeDefinition.STORE_NAME ].store;
+		select = registry.select( storeDefinition.STORE_NAME );
+
+		apiFetchSpy = jest.spyOn( { apiFetch }, 'apiFetch' );
+	} );
+
+	afterAll( () => {
+		API.setUsingCache( true );
+	} );
+
+	afterEach( () => {
+		unsubscribeFromAll( registry );
+		apiFetchSpy.mockRestore();
+	} );
+
+	describe( 'name', () => {
+		it( 'returns the correct default store name', () => {
+			expect( storeDefinition.STORE_NAME ).toEqual( `${ STORE_ARGS[ 0 ] }/${ STORE_ARGS[ 1 ] }` );
+		} );
+	} );
+
+	describe( 'actions', () => {
+		describe( 'setSettings', () => {
+			it( 'requires the values param', () => {
+				expect( () => {
+					dispatch.setSettings();
+				} ).toThrow( 'values is required.' );
+			} );
+
+			it( 'updates the respective settings', () => {
+				const values1 = { setting1: 'old', setting2: 'old' };
+				const values2 = { setting2: 'new' };
+
+				dispatch.setSettings( values1 );
+				expect( store.getState().settings ).toMatchObject( values1 );
+
+				dispatch.setSettings( values2 );
+				expect( store.getState().settings ).toMatchObject( { ...values1, ...values2 } );
+			} );
+		} );
+
+		describe( 'fetchSettings', () => {
+			it( 'does not require any params', () => {
+				expect( () => {
+					dispatch.fetchSettings();
```

Comment:
I'm surprised this doesn't cause the console to error, given it should cause a fetch request to fire. 🤷‍♂

Reply:
It actually seems not to - could this be because I call `fetchSettings` directly here (the actual call is in the control)? Not too sure about the `wp.data` internals regarding this...